ros2_foxglove_bridge/src/ros2_foxglove_bridge.cpp

```diff
@@ -446,6 +492,14 @@ class FoxgloveBridge : public rclcpp::Node {
       std::bind(&FoxgloveBridge::serviceRequest, this, request, hdl));
   }
 
+  void subscribeConnectionGraphHandler(bool subscribe) {
+    if (_subscribeGraphUpdates = subscribe) {
```

Comment:
Shouldn't this be a warning/error? Maybe we don't have the standard warning flags enabled? 😱 
```suggestion
    if ((_subscribeGraphUpdates = subscribe)) {
```

Reply:
What warning would that be?

> Maybe we don't have the standard warning flags enabled?

I'm pretty sure we do

Reply:
Clang seems to warn on this with default flags: https://godbolt.org/z/n5Pz67aaP

It's called `-Wparentheses`

Reply:
There seems to be something wrong with how compiler flags are defined

https://github.com/foxglove/ros-foxglove-bridge/blob/f4b2f68cde3cc87ad65cb95a36cd7690da34b3a5/CMakeLists.txt#L41-L49

I will investigate this when I have time. I read somewhere that tinkering with `CMAKE_CXX_FLAGS` is not the recommended way to do it.

Reply:
`-Wparentheses` is included in `-Wall` but gcc does not detect it in this specific case. Clang does, so I think I will also enable clang as the default compiler for CI